pkg/cmd/api/api.go

```diff
@@ -363,25 +399,14 @@ func processResponse(resp *http.Response, opts *ApiOptions, headersOutputStream
 		responseBody = io.TeeReader(responseBody, bodyCopy)
 	}
 
-	if opts.FilterOutput != "" && serverError == "" {
-		// TODO: reuse parsed query across pagination invocations
-		err = export.FilterJSON(opts.IO.Out, responseBody, opts.FilterOutput)
-		if err != nil {
-			return
-		}
-	} else if opts.Template != "" && serverError == "" {
-		// TODO: reuse parsed template across pagination invocations
-		err = template.Execute(responseBody)
+	if !opts.Paginate {
+		// Print the body immediately if not paginating
+		err = printResponse(responseBody, opts, isJSON, serverError != "", out, template)
+	} else {
+		_, err = io.Copy(out, responseBody)
```

Comment:
I wasn't considering that we only print if we're not paginating, so just as easily could refactor this better to handle all the printing decisions in `apiRun`. Should be a lot less branching and easier to maintain.